File: src/main/java/com/github/timo_reymann/ansible_vault_integration/execution/commandline/AnsibleCommandLineTransformer.java
package com.github.timo_reymann.ansible_vault_integration.execution.commandline;

import com.intellij.execution.configurations.GeneralCommandLine;
import com.intellij.openapi.project.Project;

public interface AnsibleCommandLineTransformer {
    String transformFileName(String input);

    default GeneralCommandLine transformCommandLine(Project project, GeneralCommandLine commandLine) {
        return commandLine;
    }
}

File: src/main/kotlin/de/timo_reymann/ansible_vault_integration/commandline/AnsibleCommandLineTransformer.kt
package de.timo_reymann.ansible_vault_integration.commandline

import com.intellij.execution.configurations.GeneralCommandLine
import com.intellij.openapi.project.Project

interface AnsibleCommandLineTransformer {
    fun transformFileName(input: String?): String?
    fun transformCommandLine(project: Project, commandLine: GeneralCommandLine): GeneralCommandLine? = commandLine
}
